### DISCUSSION TITLE
Kafası Karışık Keşiş

### DISCUSSION THREAD

**Discussion by FatihOnder81** [sonsuzus/sonsuzus#406] [Soru Cevap]
sorun
Keşiş tüm öğrencilerine sabırla ilgili bir ders vermek istiyor çünkü onlar her zaman çılgınca bir şeyler yapmak için acele ediyorlar. Onlara bunu öğretmek için, onlara farklı olabilen veya olmayabilen N sayıdan oluşan bir liste verir . Öğrencilerden bu N sayıların dizisine dayalı basit bir Matematik denklemini çözmeleri bekleniyor .

g(x) - OBEB (a[ 0 ], a[ 1 ], a[ 2 ]... a[n-1] )
f(x) - (a[ 0 ] * a[ 1 ] * a[ 2 ]... * a[n-1] )
MonkQuotient'in değeri: 10 9 + 7.

Çözülmesi gereken denklem: ( f(x) g(x) ) ) [%](https://en.wikipedia.org/wiki/Modulo_operation) MonkQuotient

Giriş kısıtlamaları: Girişin ilk satırı bir tam
sayı içerecektir - N. Bir sonraki satır listenin elemanlarını belirten N adet tamsayı içerecektir .

Çıktı kısıtlamaları:
Denklemin gerekli cevabını yazdırın.

Kısıtlamalar:
1 ≤ N ≤ 50
1 ≤ A i ≤ 10 3

**Comment by EfeMertIrez** [comment]
```
N = input()                  
arr = input().split(" ")
nums = [int(i) for i in arr]
mq = 10 ** 9 + 7
 
def f(args):
    s = 1
    for arg in args:
        s *= arg
    return s
 
def g(args):
  if len(args) > 1:
    r1, r2 = args[0], args[1]
    while r2 != 0:
        q = r1 // r2
        r1, r2 = r2, r1 - q * r2
    return g([r1] + args[2:])
  else:
    return args[0]
 
print(f(nums) ** g(nums) % mq)   
```
